sky/packages/sky/lib/src/rendering/shifted_box.dart

```diff
@@ -179,31 +180,37 @@ class RenderPositionedBox extends RenderShiftedBox {
     markNeedsLayout();
   }
 
-  ShrinkWrap _shrinkWrap;
-  ShrinkWrap get shrinkWrap => _shrinkWrap;
-  void set shrinkWrap (ShrinkWrap value) {
-    assert(value != null);
-    if (_shrinkWrap == value)
+  double _widthFactor;
+  double get widthFactor => _widthFactor;
+  void set widthFactor (double value) {
+    if (_widthFactor == value)
       return;
-    _shrinkWrap = value;
+    _widthFactor = value;
     markNeedsLayout();
   }
 
-  // These are only valid during performLayout() and paint(), since they rely on constraints which is only set after layout() is called.
-  bool get _shinkWrapWidth => _shrinkWrap == ShrinkWrap.width || _shrinkWrap == ShrinkWrap.both || constraints.maxWidth == double.INFINITY;
-  bool get _shinkWrapHeight => _shrinkWrap == ShrinkWrap.height || _shrinkWrap == ShrinkWrap.both || constraints.maxHeight == double.INFINITY;
+  double _heightFactor;
+  double get heightFactor => _heightFactor;
+  void set heightFactor (double value) {
+    if (_heightFactor == value)
+      return;
+    _heightFactor = value;
+    markNeedsLayout();
+  }
 
   void performLayout() {
     if (child != null) {
       child.layout(constraints.loosen(), parentUsesSize: true);
-      size = constraints.constrain(new Size(_shinkWrapWidth ? child.size.width : double.INFINITY,
-                                            _shinkWrapHeight ? child.size.height : double.INFINITY));
-      Offset delta = size - child.size;
+      final Size childSize = new Size(child.size.width * (_widthFactor ?? 1.0),
```

Comment:
call this desiredSize, rather than childSize, since it's not actually the child's size.